test/driver_test.rb

```diff
@@ -118,19 +118,180 @@
     it "correctly calculates the average rating" do
       trip2 = RideShare::Trip.new(
         id: 8,
-        driver: @driver,
+        driver: @rated_driver,
         passenger_id: 3,
         start_time: Time.new(2016, 8, 8),
         end_time: Time.new(2016, 8, 9),
         rating: 1
       )
-      @driver.add_trip(trip2)
+      @rated_driver.add_trip(trip2)
+
+      expect(@rated_driver.average_rating).must_be_close_to (5.0 + 1.0) / 2.0, 0.01
+    end
+
+    it "ignores in progress trips" do
+      start_rating = @rated_driver.average_rating
+
+      trip22 = RideShare::Trip.new(
+          id: 8,
+          driver: @rated_driver,
+          passenger_id: 3,
+          start_time: Time.now,
+          end_time: nil,
+          rating: nil
+      )
+      @rated_driver.assign_new_trip(trip22)
+
+      expect(@rated_driver.average_rating).must_equal start_rating
 
-      expect(@driver.average_rating).must_be_close_to (5.0 + 1.0) / 2.0, 0.01
     end
   end
 
   describe "total_revenue" do
-    # You add tests for the total_revenue method
+    before do
+      @inactive_driver = RideShare::Driver.new(
+          id: 100,
+          name: "Sad Dan",
+          vin: "12345678901234567"
+      )
+
+      @busy_driver = RideShare::Driver.new(
+          id: 40,
+          name: "Spartacus",
+          vin: "18293856382910583"
+      )
+
+      trip1 = RideShare::Trip.new(
+          id: 20,
+          driver_id: 40,
+          passenger_id: 30,
+          start_time: Time.new(2020, 7, 7, 5, 15),
+          end_time: Time.new(2020, 7, 7, 5, 45),
+          cost: 26,
+          rating: 3
+      )
+      trip2 = RideShare::Trip.new(
+          id: 50,
+          driver_id: 40,
+          passenger_id: 39,
+          start_time: Time.new(2020, 6, 7, 2, 30),
+          end_time: Time.new(2020, 6, 7, 3, 30),
+          cost: 42,
+          rating: 4
+      )
+      trip3 = RideShare::Trip.new(
+          id: 89,
+          driver_id: 40,
+          passenger_id: 206,
+          start_time: Time.new(2020, 9, 7, 8, 45),
+          end_time: Time.new(2020, 9, 7, 9, 30),
+          cost: 17,
+          rating: 2
+      )
+
+      @busy_driver.add_trip(trip1)
+      @busy_driver.add_trip(trip2)
+      @busy_driver.add_trip(trip3)
+    end
+
+    it "returns zero if driver has no trips" do
+      expect(@inactive_driver.total_revenue).must_equal 0
+    end
```

Comment:
Why does this return zero and `Passenger` raise an error? disagreement? It's important for readability that things like this are consistent!